### ISSUE TITLE
Possibly unable to raise KeplerianOrbit orbital element ValueError: duration, ecc

### ISSUE THREAD

**Issue by icweaver** [exoplanet-dev/exoplanet#213]
**Describe the bug**
Hi, amazing package and congratulations on the recent publication 🎉

I just had a quick orbital elements issue in `KeplerianOrbit` that I am running into. It was my understanding that orbits with a `duration` specified should only work for eccentric orbits:https://github.com/exoplanet-dev/exoplanet/blob/3dd400d86202537518c18998a2a7466c0541b0c8/src/exoplanet/orbits/keplerian.py#L235-L239)

I seem to be able to construct them just fine though if I specify the eccentricity manually. For example:

```python
KeplerianOrbit(duration=1.0, ecc=0.0, omega=0.0, period=2.0, t0=0.0, m_star=1.0, r_star=1.0, ror=0.01)
```

or,

```python
KeplerianOrbit(duration=1.0, ecc=0.01, omega=0.0, period=2.0, t0=0.0, m_star=1.0, r_star=1.0, ror=0.01)
```

do not seem to raise any error. If I try to satisfy the conditional by not specifying the eccentricity at all, it seems to get picked off early here:
https://github.com/exoplanet-dev/exoplanet/blob/3dd400d86202537518c18998a2a7466c0541b0c8/src/exoplanet/orbits/keplerian.py#L114-L118

If I then add in `b` and drop `m_star`, for example:

```python
KeplerianOrbit(duration=1.0, b=0.0, period=2.0, t0=0.0, r_star=1.0, ror=0.01)
```

this also constructs the orbit instead of throwing the duration error. I was wondering if this is the intended behavior, and if so what the correct usage would be to successfully raise the "fitting with duration only works for eccentric orbits" value error. Thanks! 

**To Reproduce**
```python
from exoplanet.orbits.keplerian import KeplerianOrbit

KeplerianOrbit(duration=1.0, b=0.0, period=2.0, t0=0.0, r_star=1.0, ror=0.01)

# <exoplanet.orbits.keplerian.KeplerianOrbit at blah>
```

**Expected behavior**
```python
from exoplanet.orbits.keplerian import KeplerianOrbit

KeplerianOrbit(duration=1.0, b=0.0, period=2.0, t0=0.0, r_star=1.0, ror=0.01)

# ValueError: fitting with duration only works for eccentric orbits
```

**Your setup (please complete the following information):**
 - Version of exoplanet: 0.5.2.dev2+g3dd400d
 - Operating system: Linux Mint 20.1 x86_64
 - Python version & installation method (pip, conda, etc.): pip

**Comment by dfm** [comment]
This is an excellent question! That error message is very misleading and it's actually only there for historical reasons, but this is all related to a larger issue that I don't love how this `KeplerianOrbit` works out its set of consistent parameters since it's all a bit hacky - there must be a better way!

So to explain what happens currently:

1. If `duration` is provided but `ecc` is not, the duration is used to compute the effective semimajor axis of the orbit. This will then correspond to an effective stellar density (rho_circ) for each planet which could be used for the photoeccentric effect (see, e.g. https://gallery.exoplanet.codes/tutorials/quick-tess/#bonus-eccentricity).
2. If both `duration` and `ecc` are set, these parameters (along with `omega`) are used to compute the impact parameter that would produce that duration. I believe this to be implemented correctly, but I must admit that I haven't used to tested this mode a lot...

So, this means that the `ValueError` that you're expecting will never be hit (and should probably be removed) since that case is caught earlier:

https://github.com/exoplanet-dev/exoplanet/blob/d2f8f46aa9fe7b858dba909f6c3424d9f46d4b34/src/exoplanet/orbits/keplerian.py#L111

I implemented `#1` above *after* `#2` which is why that check exists even though it is never hit.

**Comment by icweaver** [comment]
Oh, that makes a lot of sense! Thanks for the historical context and for the reference to Dawson+Johnson 2012!

I totally agree that handling all of the possible user inputs to create a self-consistent Keplerian orbit can quickly get hairy. In an earlier iteration of our Julia implementation, I played around with using multiple dispatch to deal with just a handful of inputs at first, e.g. [an orbit defined by the stellar density](https://github.com/JuliaAstro/Transits.jl/blob/1303ea0cbc083e139dc898f646397a4347687c69/src/orbits/keplerian.jl#L89) and [one defined by a/R_star](https://github.com/JuliaAstro/Transits.jl/blob/1303ea0cbc083e139dc898f646397a4347687c69/src/orbits/keplerian.jl#L113), but when attempting to generalize this to more possible combinations of inputs, things quickly blew up. I think the approach taken in your package using nested conditionals is definitely the way to go and is the most comprehensive that I have ever seen, haha. 

I'm sure that there must be better ways out there too, but I think just having random people like me trying out different edge cases like this is probably the best way to go for now!

Also, this is completely minor, but I also happened to notice a [small typo](https://github.com/exoplanet-dev/case-studies/blob/1ac3377e507c5e54731fc9e2247a3a10393d2597/docs/tutorials/quick-tess.ipynb?short_path=8c63272#L347) in the documentation you shared with me I think (impled -> implied)
